### PR TITLE
V3.5.1 fix particle switch error

### PR DESCRIPTION
**Pull request by zxx43**
Re: #

### Changelog

* 

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check // Manual trigger with `@cocos-robot run test cases` afterward

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

Attach model in beforeRender will cause error, so move that into lateUpdate
issue: https://github.com/cocos/3d-tasks/issues/12315 & https://github.com/cocos/3d-tasks/issues/12325

### CODE REVIEW

**Comment by zxx43** [comment]
@cocos-robot run test cases

**Comment by zxx43** [comment]
@cocos-robot run test cases

**Comment by fish9167** [comment]
@cocos-robot run test cases

**Comment by dogeFu** [comment]
Resolves: cocos/3d-tasks#12297

File: cocos/particle/particle-system.ts [hunk 1]
```diff
@@ -1135,20 +1160,16 @@ export class ParticleSystem extends RenderableComponent {
...
+                this.processor.detachFromScene();
                 if (this._trailModule && this._trailModule.enable) {
-                    if (!this._trailModule.getModel()?.scene) {
-                        this._trailModule._attachToScene();
-                    }
+                    this._trailModule._detachFromScene();
                 }
+                this._needSwitch = false;
+            }
+        } else if (this.getParticleCount() > 0) {
```
Comment (pandamicro) [inline cocos/particle/particle-system.ts]: Just use else is enough, the condition is contrary to the if condition

**Review by pandamicro — CHANGES_REQUESTED** [review]

**Comment by zxx43** [comment]
we have to update descriptorSet before attach，so put attach in update or lateUpdate will be fine

**Review by pandamicro — APPROVED** [review]

**Pull request merged**